unison-cli/src/Unison/Codebase/Editor/HandleInput/AuthLogin.hs

```diff
@@ -54,7 +55,10 @@ ensureAuthenticatedWithCodeserver :: CodeserverURI -> Cli UserInfo
 ensureAuthenticatedWithCodeserver codeserverURI = do
   Cli.Env {credentialManager} <- ask
   getCredentials credentialManager (codeserverIdFromCodeserverURI codeserverURI) >>= \case
-    Right (CodeserverCredentials {userInfo}) -> pure userInfo
+    Right (CodeserverCredentials {tokens = Tokens {accessToken}, userInfo}) -> do
+      isExpired accessToken >>= \case
```

Comment:
It might be worth looking into moving the `isExpired` check into `getCredentials` itself and having it return `Left (ReauthRequired ...)` there?

I think you'd need to move `isExpired` to make the imports work, but that should be fine.